exchanges/execute/src/execute.ts

```diff
@@ -0,0 +1,67 @@
+import { pipe, share, filter, map, fromPromise, mergeMap, merge } from 'wonka';
+import {
+  GraphQLSchema,
+  GraphQLFieldResolver,
+  GraphQLTypeResolver,
+  execute,
+} from 'graphql';
+import { Exchange, makeResult, makeErrorResult } from '@urql/core';
+
+interface ExecuteExchangeArgs {
+  schema: GraphQLSchema;
+  rootValue?: any;
+  contextValue?: any;
+  fieldResolver?: GraphQLFieldResolver<any, any>;
+  typeResolver?: GraphQLTypeResolver<any, any>;
+}
+
+const SUPPORTED_OPERATION_TYPES = ['query', 'mutation'];
+
+/** Exchange for executing queries locally on a schema using graphql-js. */
+export const executeExchange = ({
+  schema,
+  rootValue,
+  contextValue,
+  fieldResolver,
+  typeResolver,
+}: ExecuteExchangeArgs): Exchange => ({ forward }) => {
+  return ops$ => {
+    const sharedOps$ = share(ops$);
+
+    const executedOps$ = pipe(
+      sharedOps$,
+      filter(operation =>
+        SUPPORTED_OPERATION_TYPES.includes(operation.operationName)
+      ),
+      map(async operation => {
+        try {
+          const result = await execute(
+            schema,
+            operation.query,
+            rootValue,
+            typeof contextValue === 'function' ? contextValue() : contextValue,
```

Comment:
I think the only thing that's left here is provide some useful data to `contextValue()`; I think we could just pass `operation` or `operation.context`. Mainly this will need to give the user access to `operation.context` so that options like `operation.context.fetchOptions` are accessible.

Also, I think, I mentioned this in a nother comment, but some implementations, like `express-graphql` choose to name this option `context` instead of `contextValue`. That could potentially be confusing, since we already have `operation.context`, but technically it's not only a value anymore given that it can be computed: https://graphql.org/graphql-js/express-graphql/

Reply:
Now called as `context(operation)`